src/main/groovy/com/google/protobuf/gradle/ProtobufExtract.groovy

```diff
@@ -58,80 +58,92 @@ import javax.inject.Inject
 @CompileStatic
 abstract class ProtobufExtract extends DefaultTask {
 
-  /**
-   * The directory for the extracted files.
-   */
-  private File destDir
-  private Boolean isTest = null
-  private final ConfigurableFileCollection inputFiles = objectFactory.fileCollection()
   private final CopyActionFacade copyActionFacade = instantiateCopyActionFacade()
   private final ArchiveActionFacade archiveActionFacade = instantiateArchiveActionFacade()
   private final FileCollection filteredProtos = instantiateFilteredProtos()
 
-  public void setIsTest(boolean isTest) {
-    this.isTest = isTest
-  }
+  @OutputDirectory
+  public abstract DirectoryProperty getDestDir()
 
-  @Input
-  public boolean getIsTest() {
-    Preconditions.checkNotNull(isTest)
-    return isTest
-  }
+  // Used to configure ide proto sources
+  @Internal
+  public abstract Property<Boolean> getIsTest()
 
+  // Input tracked in getInputProtoFiles method
   @Internal
-  public ConfigurableFileCollection getInputFiles() {
-    return inputFiles
-  }
+  public abstract ConfigurableFileCollection getInputFiles()
 
-  /**
-   * Inputs for this task containing only proto files, which is enough for up-to-date checks.
-   * Add inputs to inputFiles. Uses relative path sensitivity as directory layout changes impact output.
-   */
+  // Inputs for this task containing only proto files, which is enough for up-to-date checks.
+  // Add inputs to inputFiles. Uses relative path sensitivity as directory layout changes impact output.
   @InputFiles
   @PathSensitive(PathSensitivity.RELATIVE)
-  FileTree getFilteredProtosFromInputs() {
-    return filteredProtos.asFileTree.matching { PatternFilterable pattern -> pattern.include("**/*.proto") }
-  }
-
-  @Internal
-  CopyActionFacade getCopyActionFacade() {
-    return copyActionFacade
-  }
-
-  @Internal
-  ArchiveActionFacade getArchiveActionFacade() {
-    return archiveActionFacade
+  public FileTree getInputProtoFiles() {
```

Comment:
Why this change?

Reply:
IMO `getInputProtoFiles` better name than `getFilteredProtosFromInputs`.

Reply:
Okay, that's fine.